File: java/src/main/java/com/adaptionsoft/games/uglytrivia/domain/GameServices.java
package com.adaptionsoft.games.uglytrivia.domain;

import java.util.Random;

public class GameServices implements GameFacade {
    private final Games games;
    private final Questions questions;
    private final EventsPublisher output;

    public GameServices(Games games, Questions questions, EventsPublisher output) {
        this.games = games;
        this.questions = questions;
        this.output = output;
    }

    @Override
    public int newGame() {
        Game game = new Game(questions);
        games.save(game);
        return 0;
    }

    @Override
    public void addPlayer(int gameId, String playerName){
        Game game = games.get(gameId);
        game.add(playerName);
        games.save(game);
    }

    @Override
    public void roll(int gameId){
        Game game = games.get(gameId);
        Random random = new Random();
        game.roll(random.nextInt(5) + 1);
        games.save(game);
    }
}

File: php/src/Domain/GameServices.php
<?php


namespace Trivia\Domain;


class GameServices implements GameFacade
{
    /**
     * @var GameRepository
     */
    private $gameRepository;
    /**
     * @var \Closure
     */
    private $println;

    public function __construct(GameRepository $gameRepository, \Closure $println)
    {
        $this->gameRepository = $gameRepository;
        $this->println = $println;
    }

    function newGame(): string
    {
        $game = new Game($this->println);
        $this->gameRepository->save($game);
        return uniqid();
    }

    function addPlayer(string $gameId, string $playerName)
    {
        $game = $this->gameRepository->get($gameId);
        $game->add($playerName);
        $this->gameRepository->save($game);
    }

    function roll(string $gameId)
    {
        $game = $this->gameRepository->get($gameId);
        $game->roll(rand(0, 5) + 1);
        $this->gameRepository->save($game);
    }
}